File: solid-demo/I-Principle-CONFORMED/Models/Customer/Customer.java
package Customer;

import Product.Product;

import java.util.List;

public class Customer {
    private String name;
    private int id;
    private List<Product> productsBought;
    private boolean chooseDeliveryService;

    public Customer(String name, int id, List<Product> productsBought, boolean chooseDeliveryService) {
        this.name = name;
        this.id = id;
        this.productsBought = productsBought;
        this.chooseDeliveryService = chooseDeliveryService;
    }

    public String getName(){
        return name;
    }

    public boolean getDeliveryChoice(){
        return chooseDeliveryService;
    }
}

File: solid-demo/I-Principle-VIOLATED/Models/Customer/Customer.cs
using System.Collections.Generic;

namespace I_Principle_VIOLATED.Models
{
    public class Customer 
    {
        private string name;
        private int id;
        private IEnumerable<Product> productsBought;
        private bool chooseDeliveryService;
        IServiceOperation pickService;
        public Customer(string name, int id, IEnumerable<Product> productsBought, bool chooseDeliveryService) {
            this.name = name;
            this.id = id;
            this.productsBought = productsBought;
            this.chooseDeliveryService = chooseDeliveryService;
        }

        public string getName(){
            return name;
        }
        public bool getDeliveryChoice(){
            return chooseDeliveryService;
        }
    }   

}
